src/components/cc-product-list/products-controller.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @typedef {import('./cc-product-list.types.js').ProductsCategory} ProductsByCategory
+ * @typedef {import('./cc-product-list').CcProductList} CcProductList
+ * @typedef {import('./cc-product-list.types.js').Product} Product
+ * @typedef {import('./cc-product-list.types.js').CategoryFilter} CategoryFilter
+ */
+
+export class ProductsController {
+
+  /**
+   * @param {CcProductList} host
+   */
+  constructor (host) {
+    /** @type {CcProductList} The host. */
+    this._host = host;
+
+    /** @type {ProductsByCategory[]} The initial products category list. */
+    this._productsByCategories = [];
+
+    /** @type {CategoryFilter[]} Categories and their current state.  */
+    this._categoriesFilters = [];
+
+    /** @type {string|null} Current category selected.  */
+    this._currentCategoryNameFilter = null;
```

Comment:
praise: I was about to say "what about juste `currentCategoryFilter` but we would be confused about whether it's a category object or a string.